ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils_filtered_tenants.py

```diff
@@ -0,0 +1,221 @@
+        """
+        Return switch.filter_tags + switch.group if defined
+        """
+        filter_tags = get(self._hostvars, "switch.filter_tags", default=[])
+        if (switch_group := get(self._hostvars, "switch.group")) is not None:
+            filter_tags.append(switch_group)
+
+        return filter_tags
+
+    def _filtered_vrfs(self, tenant: dict) -> list[dict]:
+        """
+        Return sorted and filtered vrf list from given tenant.
+        Filtering based on svi tags, l3interfaces and switch.always_include_vrfs_in_tenants.
+        Keys of VRF data model will be converted to lists.
+        """
+        filtered_vrfs = []
+        always_include_vrfs_in_tenants = get(self._hostvars, "switch.always_include_vrfs_in_tenants", default=[])
+
+        vrfs: list[dict] = natural_sort(convert_dicts(tenant.get("vrfs", []), "name"), "name")
+        for vrf in vrfs:
+            bgp_peers = natural_sort(convert_dicts(vrf.get("bgp_peers"), "ip_address"), "ip_address")
+            vrf["bgp_peers"] = [bgp_peer for bgp_peer in bgp_peers if self._hostname in bgp_peer.get("nodes", [])]
+            vrf["static_routes"] = [route for route in vrf.get("static_routes", []) if self._hostname in route.get("nodes", [self._hostname])]
+            vrf["ipv6_static_routes"] = [route for route in vrf.get("ipv6_static_routes", []) if self._hostname in route.get("nodes", [self._hostname])]
+            vrf["svis"] = self._filtered_svis(vrf)
+            vrf["l3_interfaces"] = [
+                l3_interface
+                for l3_interface in vrf.get("l3_interfaces", [])
+                if (
+                    self._hostname in l3_interface.get("nodes", [])
+                    and l3_interface.get("ip_addresses") is not None
+                    and l3_interface.get("interfaces") is not None
+                )
+            ]
+            if vrf["svis"] or vrf["l3_interfaces"] or "all" in always_include_vrfs_in_tenants or tenant["name"] in always_include_vrfs_in_tenants:
+                filtered_vrfs.append(vrf)
+
+            vrf["additional_route_targets"] = [
+                rt
+                for rt in vrf.get("additional_route_targets", [])
+                if (
+                    self._hostname in rt.get("nodes", [self._hostname])
+                    and rt.get("address_family") is not None
+                    and rt.get("route_target") is not None
+                    and rt.get("type") in ["import", "export"]
+                )
+            ]
+
+        return filtered_vrfs
+
+    @cached_property
+    def _svi_profiles(self) -> list[dict]:
+        """
+        Return list of svi_profiles
+        """
+        svi_profiles = convert_dicts(get(self._hostvars, "svi_profiles", default=[]), "profile")
+        for svi_profile in svi_profiles:
+            svi_profile["nodes"] = self._filtered_svi_nodes(svi_profile)
+        return svi_profiles
+
+    def _filtered_svi_nodes(self, svi: dict) -> list[dict]:
```

Comment:
Given that we are fitering can we just not return the node_config and get rid of the list (and all the [0] access) ?

Reply:
This will go away before we merge this, since we now combine the node config onto the svi - supporting setting any key under node.